apps/rush-lib/src/logic/PublishGit.ts

```diff
@@ -4,48 +4,62 @@
 import { PublishUtilities } from './PublishUtilities';
 import { Utilities } from '../utilities/Utilities';
 import { RushConfigurationProject } from '../api/RushConfigurationProject';
+import { Git } from './Git';
 
 export class PublishGit {
-  private _targetBranch: string | undefined;
+  private readonly _targetBranch: string;
+  private readonly _gitPath: string;
 
-  public constructor(targetBranch: string | undefined) {
+  public constructor(git: Git, targetBranch: string) {
     this._targetBranch = targetBranch;
+    this._gitPath = git.getGitPathOrThrow();
   }
 
-  public checkout(branchName: string | undefined, createBranch?: boolean): void {
-    const params: string = `checkout ${createBranch ? '-b ' : ''}${branchName}`;
+  public checkout(branchName: string, createBranch: boolean = false): void {
+    const params: string[] = ['checkout'];
+    if (createBranch) {
+      params.push('-b');
+    }
+
+    params.push(branchName);
 
-    PublishUtilities.execCommand(!!this._targetBranch, 'git', params.split(' '));
+    PublishUtilities.execCommand(!!this._targetBranch, this._gitPath, params);
   }
 
   public merge(branchName: string): void {
-    PublishUtilities.execCommand(!!this._targetBranch, 'git', `merge ${branchName} --no-edit`.split(' '));
+    PublishUtilities.execCommand(!!this._targetBranch, this._gitPath, ['merge', branchName, '--no-edit']);
   }
 
   public deleteBranch(branchName: string, hasRemote: boolean = true): void {
-    PublishUtilities.execCommand(!!this._targetBranch, 'git', `branch -d ${branchName}`.split(' '));
+    PublishUtilities.execCommand(!!this._targetBranch, this._gitPath, ['branch', '-d', branchName]);
     if (hasRemote) {
-      PublishUtilities.execCommand(
-        !!this._targetBranch,
-        'git',
-        `push origin --delete ${branchName}`.split(' ')
-      );
+      PublishUtilities.execCommand(!!this._targetBranch, this._gitPath, [
+        'push',
+        'origin',
+        '--delete',
+        branchName
```

Comment:
This PR is changing `rush publish` code. Seems possible that it could introduce a regression that might not be noticed right away.

Can you do some kind of basic smoke test of the affected code paths?

Reply:
Tested publishing.